typhon/cloudmask/aster.py

```diff
@@ -542,116 +710,81 @@ def lapserate_moist_adiabate():
-    ast_azimuth[np.logical_and(n_az<0, ~np.isnan(n))] = 90 + S
-    # swath left side
-    ast_azimuth[np.logical_and(n_az>=0, ~np.isnan(n))] = 270 + S
-
-    # Reflection angle, i.e. the angle between the sensor and the reflected sun
-    # in the sensor-sun-plane.
-    reflection_angle = theta_r(sun_zenith=sun_zenith,
-                               # +180° corresponds to "anti-/reflected" sun
-                               sun_azimuth=sun_azimuth + 180,
-                               sensor_zenith=ast_zenith,
-                               sensor_azimuth=ast_azimuth)
+    elif method=='modis':
+        lapserate = lapserate_modis(month, latitude)
+    else:
+        raise ValueError("Method is not supported.")
 
-    return reflection_angle
+    resolution_ratio = np.shape(cloudmask)[0] // np.shape(bt)[0]
 
+    cloudmask_inverted = cloudmask.copy()
+    cloudmask_inverted[np.isnan(cloudmask_inverted)] = 1
+    cloudmask_inverted = np.asarray(np.invert(np.asarray(
+                                cloudmask_inverted, dtype=bool)), dtype=int)
 
-def theta_r(sun_zenith, sun_azimuth, sensor_zenith, sensor_azimuth):
-    """Calculate the reflected sun angle, theta_r, of specular reflection
-    of sunlight into an instrument sensor.
+    cloudmask[np.isnan(cloudmask)] = 0
+    cloudmask = np.asarray(cloudmask, dtype=int)
 
-    Parameters:
-        sun_zenith (float): sun zenith angle in [°].
-        sun_azimuth (float): sun azimuth angle in [°].
-        sensor_zenith (ndarray): 2D sensor zenith angle in [°] for eachannel pixel
-                                in the image.
-        sensor_azimuth (ndarray): 2D sensor azimuth angle in [°] for eachannel
-                                pixel in the image.
+    # Match resolutions of cloud mask and brightness temperature (bt) arrays.
+    if resolution_ratio > 1:
+        # On bt resolution, flag pixels as cloudy only if all subgrid pixels
+        # are cloudy in the original cloud mask.
+        mask_cloudy = block_reduce(cloudmask,
+                                        (resolution_ratio, resolution_ratio),
+                                        func=np.alltrue)
+        # Search for only clear pixels to derive a bt clearsky/ocean value.
+        mask_clear = block_reduce(cloudmask_inverted,
+                                        (resolution_ratio, resolution_ratio),
+                                        func=np.alltrue)
+    elif resolution_ratio < 1:
+        try:
+            mask_cloudy = np.repeat(np.repeat(cloudmask, resolution_ratio,
+                                        axis=0), resolution_ratio, axis=1)
+            mask_clear = np.repeat(np.repeat(cloudmask_inverted,
+                                                  resolution_ratio, axis=0),
+                                                  resolution_ratio, axis=1)
+        except ValueError:
+            logger.warning(
+                'Problems matching the shapes of provided cloud mask and bt arrays.')
```

Comment:
I would recommend to raise an exception at that point. If the program enters the `except`-clause, the variables `mask_cloudy` and `mask_clear` are undefined and the program will crash later. I did not change it, as I am not sure what may cause the `try`-block to fail 😉